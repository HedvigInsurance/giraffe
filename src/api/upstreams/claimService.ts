import {HttpClient} from '../httpClient'

export interface ClaimServiceClient {
  getMemberClaims(onlyRecent: Boolean): Promise<ClaimDto[]>
}

export interface ClaimDto {
  id: string
  status: ClaimStatusDto
  contractId?: string
  type?: string
  outcome?: ClaimOutcomeDto
  registrationDate: string
  closedAt?: string
  payout?: {
    amount: string,
    currency: string
  }
  files: File[]
  audioURL?: string
}

export type File = {
  signedUrl: string
  key: string
  bucket: string
};

export enum ClaimStatusDto {
  CREATED = 'CREATED',
  IN_PROGRESS = 'IN_PROGRESS',
  CLOSED = 'CLOSED',
  REOPENED = 'REOPENED',

}

export enum ClaimOutcomeDto {
  PAID = 'PAID',
  NOT_COMPENSATED = 'NOT_COMPENSATED',
  NOT_COVERED = 'NOT_COVERED'
}


export const createClaimServiceClient = (client: HttpClient): ClaimServiceClient => {
  return {
    getMemberClaims: async (onlyRecent: Boolean) => {
      const res = await client.get(`/claims?onlyRecent=${onlyRecent}`)
      return res.json()
    }
  }
}
