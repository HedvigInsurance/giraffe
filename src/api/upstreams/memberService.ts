
import { HttpClient } from '../httpClient'

export interface MemberServiceClient {
  getSelfMember(): Promise<MemberDto>
}

export interface MemberDto {
  firstName: string
  lastName: string
  memberId: string
  email: string
  phoneNumber: string
  ssn: string
  birthDate: string
}

export const createMemberServiceClient = (client: HttpClient): MemberServiceClient => {
  return {
    getSelfMember: async () => {
      const res = await client.get("/me")
      return res.json()
    }
  }
}
