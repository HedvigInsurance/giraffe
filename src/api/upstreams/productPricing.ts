
import { HttpClient } from '../httpClient'

export interface ProductPricingClient {
  getMemberContracts(): Promise<ContractDto[]>
  getContractMarketInfo(): Promise<ContractMarketInfoDto>
  getSelfChangeEligibility(): Promise<SelfChangeEligibilityDto>
}

export interface ContractMarketInfoDto {
  market: string,
  preferredCurrency: string
}

export interface SelfChangeEligibilityDto {
  blockers: string[]
}

export interface ContractDto {
  id: string
  holderMemberId: string
  switchedFrom?: string
  masterInception?: string
  status: ContractStatusDto
  typeOfContract: string
  isTerminated: boolean
  terminationDate?: string
  currentAgreementId: string
  hasPendingAgreement: boolean
  agreements: AgreementDto[]
  upcomingAgreement?: AgreementDto
  renewal?: RenewalDto
  createdAt: string
}

export interface AgreementType {
  id: string
  fromDate?: string
  toDate?: string
  basePremium: {
    amount: string,
    currency: string
  }
  certificateUrl?: string
  status: AgreementStatusDto
}

export type AgreementDto = 
  SwedishApartmentAgreementDto |
  SwedishHouseAgreementDto |
  NorwegianHomeContentAgreementDto |
  NorwegianTravelAgreementDto |
  DanishHomeContentAgreementDto |
  DanishAccidentAgreementDto |
  DanishTravelAgreementDto

export interface SwedishApartmentAgreementDto extends AgreementType {
  type: "SwedishApartment"
  lineOfBusiness: string
  address: AddressDto
  numberCoInsured: number
  squareMeters: number
}
export interface SwedishHouseAgreementDto extends AgreementType {
  type: "SwedishHouse"
  address: AddressDto,
  numberCoInsured: number,
  squareMeters: number,
  ancillaryArea: number,
  yearOfConstruction: number,
  numberOfBathrooms: number,
  extraBuildings: ExtraBuildingDto[],
  isSubleted: boolean
}
export interface NorwegianHomeContentAgreementDto extends AgreementType {
  type: "NorwegianHomeContent"
  lineOfBusiness: string
  address: AddressDto
  numberCoInsured: number
  squareMeters: number
}
export interface NorwegianTravelAgreementDto extends AgreementType {
  type: "NorwegianTravel"
  lineOfBusiness: string
  numberCoInsured: number
}
export interface DanishHomeContentAgreementDto extends AgreementType {
  type: "DanishHomeContent"
  address: AddressDto,
  numberCoInsured: number,
  squareMeters: number,
  lineOfBusiness: string
}
export interface DanishAccidentAgreementDto extends AgreementType {
  type: "DanishAccident"
  address: AddressDto,
  numberCoInsured: number,
  lineOfBusiness: string
}
export interface DanishTravelAgreementDto extends AgreementType {
  type: "DanishTravel"
  address: AddressDto,
  numberCoInsured: number,
  lineOfBusiness: string
}

export interface RenewalDto {
  renewalDate: string
  draftCertificateUrl?: string
  draftOfAgreementId?: string
}

export enum ContractStatusDto {
  PENDING = 'PENDING',
  ACTIVE_IN_FUTURE = 'ACTIVE_IN_FUTURE',
  ACTIVE_IN_FUTURE_AND_TERMINATED_IN_FUTURE = 'ACTIVE_IN_FUTURE_AND_TERMINATED_IN_FUTURE',
  ACTIVE = 'ACTIVE',
  TERMINATED_TODAY = 'TERMINATED_TODAY',
  TERMINATED_IN_FUTURE = 'TERMINATED_IN_FUTURE',
  TERMINATED = 'TERMINATED'
}

export enum AgreementStatusDto {
  PENDING = 'PENDING',
  ACTIVE_IN_FUTURE = 'ACTIVE_IN_FUTURE',
  ACTIVE = 'ACTIVE',
  ACTIVE_IN_PAST = 'ACTIVE_IN_PAST',
  TERMINATED = 'TERMINATED'
}

export interface AddressDto {
  street: string
  coLine?: string
  postalCode: string
  city?: string
  country: string
  apartment?: string
  floor?: string
}

export interface ExtraBuildingDto {
  type: ExtraBuildingTypeDto
  area: number
  displayName: string
  hasWaterConnected: boolean
}

export enum ExtraBuildingTypeDto {
  GARAGE = 'GARAGE',
  CARPORT = 'CARPORT',
  SHED = 'SHED',
  STOREHOUSE = 'STOREHOUSE',
  FRIGGEBOD = 'FRIGGEBOD',
  ATTEFALL = 'ATTEFALL',
  OUTHOUSE = 'OUTHOUSE',
  GUESTHOUSE = 'GUESTHOUSE',
  GAZEBO = 'GAZEBO',
  GREENHOUSE = 'GREENHOUSE',
  SAUNA = 'SAUNA',
  BARN = 'BARN',
  BOATHOUSE = 'BOATHOUSE',
  OTHER = 'OTHER'
}

export const createProductPricingClient = (client: HttpClient): ProductPricingClient => {
  return {
    getMemberContracts: async () => {
      const res = await client.get("/contracts")
      return res.json()
    },
    getContractMarketInfo: async () => {
      const res = await client.get("/contracts/market-info")
      return res.json()
    },
    getSelfChangeEligibility: async () => {
      const res = await client.get("/contracts/selfChange/eligibility")
      return res.json()
    }
  }
}
