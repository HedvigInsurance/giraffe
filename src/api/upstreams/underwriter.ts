
import { HttpClient } from '../httpClient'

export interface UnderwriterClient {
  createQuote(body: CreateQuoteDto): Promise<QuoteCreationResult>
}

export interface CreateQuoteDto {
  memberId: String
  firstName: string
  lastName: string
  ssn: string
  birthDate: string
  startDate: string
  swedishApartmentData?: {
    street: string
    zipCode: string
    householdSize: number
    livingSpace: number
    subType: string
  }
  swedishHouseData?: {
    street: string
    zipCode: string
    householdSize: number
    livingSpace: number
    ancillaryArea: number
    yearOfConstruction: number
    numberOfBathrooms: number
    floor: number
    subleted: boolean
    extraBuildings: {
      type: string
      area: number
      hasWaterConnected: boolean
    }[]
  }
  norwegianHomeContentsData?: {
    street: string
    zipCode: string
    coInsured: number
    livingSpace: number
    youth: boolean
    subType: string
  }
  norwegianTravelData?: {
    coInsured: number
    youth: boolean
  }
  danishHomeContentsData?: {
    street: string
    zipCode: string
    livingSpace: number
    coInsured: number
    student: boolean
    subType: string
  }
  danishAccidentData?: {
    street: string
    zipCode: string
    coInsured: number
    student: boolean
  }
  danishTravelData?: {
    street: string
    zipCode: string
    coInsured: number
    student: boolean
  }
}

export type QuoteCreationResult = QuoteCreationSuccessDto | QuoteCreationFailureDto

export interface QuoteCreationSuccessDto {
  status: 'success',
  id: string
  price: number
  validTo: string
}

export interface QuoteCreationFailureDto {
  status: 'failure',
  breachedUnderwritingGuidelines: BreachedGuidelineDto[]
}

export interface BreachedGuidelineDto {
  code: string
}

export const createUnderwriterClient = (client: HttpClient): UnderwriterClient => {
  return {
    createQuote: async (body) => {
      // 422 means underwriting guidelines breached, and we should read the body
      const res = await client.post("/quotes", body, { validStatusCodes: [422] })
      const result = await res.json()
      if (res.status === 422) {
        return { status: 'failure', ...result}
      }
      return { status: 'success', ...result}
    }
  }
}
