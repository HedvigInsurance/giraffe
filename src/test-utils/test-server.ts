
import { Upstream } from './../api/upstream';
import { resolve } from 'path';
import { readFileSync } from 'fs'
import { gql, GraphQLUpload, ApolloServer } from 'apollo-server-koa';
import { IResolvers, makeExecutableSchema } from 'graphql-tools';
import { Context } from '../context';
import { resolvers } from '../resolvers'
import { ApolloServerTestClient, createTestClient } from 'apollo-server-testing'
import { RedisPubSub } from 'graphql-redis-subscriptions';

const typeDefs = gql(
  readFileSync(resolve(__dirname, '../schema.graphqls'), 'utf8'),
)

const upstream: Upstream = {
  productPricing: {
    getContract: () => Promise.reject("getContract Not implemented"),
    getMemberContracts: () => Promise.reject("getMemberContracts Not implemented"),
    getContractMarketInfo: () => Promise.reject("getContractMarketInfo Not implemented"),
    getSelfChangeEligibility: () => Promise.reject("getSelfChangeEligibility Not implemented")
  },
  underwriter: {
    createQuote: (_) => Promise.reject("createQuote Not implemented")
  },
  memberService: {
    getSelfMember: () => Promise.reject("getSelfMember Not implemented")
  }
}

const context: Context = {
    getToken: () => 'test-token',
    headers: {
      'User-Agent': 'Testing',
      'X-Forwarded-For': 'Testing',
      'X-Request-Id': 'Testing',
      'Accept-Language': 'en',
      'Enable-Simple-Sign': 'false',
    },
    remoteIp: '127.0.0.1',
    upstream: upstream,
    pubsub: {} as RedisPubSub, // fake this at some point
    strings: (key) => key
}

const localSchema = makeExecutableSchema<Context>({
  typeDefs,
  // @ts-ignore - This type is incorrect
  resolvers: {
    Upload: GraphQLUpload,
    ...(resolvers as IResolvers<any, Context>),
  },
})

export interface TestingContext {
  query: ApolloServerTestClient['query'],
  mutate: ApolloServerTestClient['mutate'],
  upstream: Upstream,
  stop(): Promise<void>
}

export const startApolloTesting = (): TestingContext => {
  const server = new ApolloServer({
    schema: localSchema,
    context: context
  })
  const { query, mutate } = createTestClient(server)
  return {
    query,
    mutate,
    upstream,
    stop: server.stop
  }
}
