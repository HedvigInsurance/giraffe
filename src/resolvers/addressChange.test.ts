import { ContractStatusDto } from '../api/upstreams/productPricing'
import gql from 'graphql-tag'
import { ContractMarketInfoDto } from '../api/upstreams/productPricing'
import { MemberDto } from '../api/upstreams/memberService'
import { startApolloTesting } from '../test-utils/test-server'

const { mutate, upstream } = startApolloTesting()

const createQuote = jest.fn().mockResolvedValue('qid1')
beforeEach(() => {
  createQuote.mockClear()
  createQuote.mockResolvedValue('qid1')

  upstream.memberService.getSelfMember = () => Promise.resolve(member)
  upstream.underwriter.createQuote = createQuote
})

// Tests

describe('createAddressChangeQuotes - SE_APARTMENT', () => {
  beforeEach(() => {
    mockMarket('SWEDEN')
    mockContracts(['SE_APARTMENT'])
  })

  it('non student rental', async () => {
    await MUTATIONS.createAddressChangeQuotes(seNonStudentRental)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
  it('student rental', async () => {
    await MUTATIONS.createAddressChangeQuotes(seStudentRental)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
  it('non student owned', async () => {
    await MUTATIONS.createAddressChangeQuotes(seNonStudentOwned)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
  it('student owned', async () => {
    await MUTATIONS.createAddressChangeQuotes(seStudentOwned)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
})

describe('createAddressChangeQuotes - SE_HOUSE', () => {
  beforeEach(() => {
    mockMarket('SWEDEN')
    mockContracts(['SE_HOUSE'])
  })

  it('works', async () => {
    await MUTATIONS.createAddressChangeQuotes(seHouse)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
})

describe('createAddressChangeQuotes - Norway', () => {
  beforeEach(() => {
    mockMarket('NORWAY')
    mockContracts(['NO_HOME_CONTENT_YOUTH_OWN', 'NO_TRAVEL_YOUTH'])
  })

  it('works', async () => {
    await MUTATIONS.createAddressChangeQuotes(noHomeContent)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
})

describe('createAddressChangeQuotes - Denmark', () => {
  beforeEach(() => {
    mockMarket('DENMARK')
    mockContracts(['DK_HOME_CONTENT_OWN', 'DK_TRAVEL', 'DK_ACCIDENT'])
  })

  it('works', async () => {
    await MUTATIONS.createAddressChangeQuotes(dkHomeContent)
    expect(createQuote.mock.calls).toMatchSnapshot()
  })
})

// Helpers

const MUTATIONS = {
  createAddressChangeQuotes: async (input: any) =>
    await mutate({
      mutation: gql`
        mutation($input: AddressChangeInput!) {
          createAddressChangeQuotes(input: $input) {
            ... on AddressChangeQuoteSuccess {
              quoteIds
            }
            ... on AddressChangeQuoteFailure {
              breachedUnderwritingGuidelines
            }
          }
        }
      `,
      variables: { input },
    }),
}

const mockMarket = (market: string) => {
  upstream.productPricing.getContractMarketInfo = () => Promise.resolve({
    ...marketInfo,
    market,
  })
}
  
const mockContracts = (typesOfContract: string[]) => {
  const mock = jest.fn()
  typesOfContract.forEach((typeOfContract, index) =>
    mock.mockResolvedValueOnce({
      id: `cid${index}`,
      status: ContractStatusDto.ACTIVE,
      typeOfContract,
    })
  )
  upstream.productPricing.getContract = mock
}

const member: MemberDto = {
  firstName: 'Test',
  lastName: 'Testsson',
  memberId: '123',
  email: 'test@test.com',
  phoneNumber: '070123456789',
  ssn: '201212121212',
  birthDate: '1991-07-27',
}

const marketInfo: ContractMarketInfoDto = {
  market: 'UNKNOWN',
  preferredCurrency: 'UNKNOWN',
}

const baseInput = {
  contractBundleId: 'bundle-abc',
  startDate: '2021-06-01',
  street: 'Fakestreet 123',
  zip: '12345',
  numberCoInsured: 2,
  livingSpace: 44,
}

const seNonStudentRental = {
  ...baseInput,
  type: 'APARTMENT',
  ownership: 'RENT',
  isStudent: false,
}

const seStudentRental = {
  ...baseInput,
  type: 'APARTMENT',
  ownership: 'RENT',
  isStudent: true,
}

const seNonStudentOwned = {
  ...baseInput,
  type: 'APARTMENT',
  ownership: 'BRF',
  isStudent: false,
}

const seStudentOwned = {
  ...baseInput,
  type: 'APARTMENT',
  ownership: 'BRF',
  isStudent: true,
}

const seHouse = {
  ...baseInput,
  type: 'HOUSE',
  ancillaryArea: 28,
  yearOfConstruction: 1912,
  numberOfBathrooms: 3,
  isSubleted: true,
  extraBuildings: [
    {
      type: 'GARAGE',
      area: 12,
      hasWaterConnected: false,
    },
  ],
  ownership: 'OWN',
  isStudent: false,
}

const noHomeContent = {
  ...baseInput,
  contractBundleId: 'bundle-abc,def', // two contracts
  type: 'APARTMENT',
  ownership: 'OWN',
  isYouth: true,
}

const dkHomeContent = {
  ...baseInput,
  contractBundleId: 'bundle-abc,def,ghi', // three contracts
  type: 'APARTMENT',
  ownership: 'OWN',
  isStudent: false,
}
