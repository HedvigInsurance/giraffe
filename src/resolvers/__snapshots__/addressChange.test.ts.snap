// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAddressChangeQuotes - Denmark works 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "danishHomeContentsData": Object {
        "coInsured": 2,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "student": false,
        "subType": "OWN",
        "zipCode": "12345",
      },
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
    },
  ],
  Array [
    Object {
      "birthDate": "1991-07-27",
      "danishTravelData": Object {
        "coInsured": 2,
        "street": "Fakestreet 123",
        "student": false,
        "zipCode": "12345",
      },
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
    },
  ],
  Array [
    Object {
      "birthDate": "1991-07-27",
      "danishAccidentData": Object {
        "coInsured": 2,
        "street": "Fakestreet 123",
        "student": false,
        "zipCode": "12345",
      },
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
    },
  ],
]
`;

exports[`createAddressChangeQuotes - Norway works 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "norwegianHomeContentsData": Object {
        "coInsured": 2,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "subType": "OWN",
        "youth": true,
        "zipCode": "12345",
      },
      "ssn": "201212121212",
      "startDate": "2021-06-01",
    },
  ],
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "norwegianTravelData": Object {
        "coInsured": 2,
        "youth": true,
      },
      "ssn": "201212121212",
      "startDate": "2021-06-01",
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_APARTMENT non student owned 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishApartmentData": Object {
        "householdSize": 3,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "subType": "BRF",
        "zipCode": "12345",
      },
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_APARTMENT non student rental 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishApartmentData": Object {
        "householdSize": 3,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "subType": "RENT",
        "zipCode": "12345",
      },
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_APARTMENT student owned 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishApartmentData": Object {
        "householdSize": 3,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "subType": "BRF_STUDENT",
        "zipCode": "12345",
      },
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_APARTMENT student rental 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishApartmentData": Object {
        "householdSize": 3,
        "livingSpace": 44,
        "street": "Fakestreet 123",
        "subType": "RENT_STUDENT",
        "zipCode": "12345",
      },
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_HOUSE accepts undefined for \`extraBuildings\` 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishHouseData": Object {
        "ancillaryArea": 28,
        "extraBuildings": Array [],
        "householdSize": 3,
        "livingSpace": 44,
        "numberOfBathrooms": 3,
        "street": "Fakestreet 123",
        "subleted": true,
        "yearOfConstruction": 1912,
        "zipCode": "12345",
      },
    },
  ],
]
`;

exports[`createAddressChangeQuotes - SE_HOUSE works 1`] = `
Array [
  Array [
    Object {
      "birthDate": "1991-07-27",
      "firstName": "Test",
      "initiatedFrom": "SELF_CHANGE",
      "lastName": "Testsson",
      "memberId": "123",
      "ssn": "201212121212",
      "startDate": "2021-06-01",
      "swedishHouseData": Object {
        "ancillaryArea": 28,
        "extraBuildings": Array [
          Object {
            "area": 12,
            "hasWaterConnected": false,
            "type": "GARAGE",
          },
        ],
        "householdSize": 3,
        "livingSpace": 44,
        "numberOfBathrooms": 3,
        "street": "Fakestreet 123",
        "subleted": true,
        "yearOfConstruction": 1912,
        "zipCode": "12345",
      },
    },
  ],
]
`;
